### ISSUE TITLE
clang-format does not handle byte order marker / utf-16

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#29053]
|  |  |
| --- | --- |
| Bugzilla Link | [28679](https://llvm.org/bz28679) |
| Version | 3.8 |
| OS | Linux |
| Attachments | [File with BOM marker. Possible that the upload stripped it out though; first two bytes should be 0xff 0xfe.](https://user-images.githubusercontent.com/60944935/143753986-48ce0df3-4f8e-458e-95e1-1e18fdb0c0d6.gz) |
| Reporter | LLVM Bugzilla Contributor |

## Extended Description 

$ clang-format -style="google" -i break.h
0  libLLVM-3.8.so.1 0x00007f714174bd38 llvm::sys::PrintStackTrace(llvm::raw_ostream&) + 56
1  libLLVM-3.8.so.1 0x00007f7141749fc6 llvm::sys::RunSignalHandlers() + 54
2  libLLVM-3.8.so.1 0x00007f714174a129
3  libc.so.6        0x00007f7140a9d4a0
4  clang-format     0x000000000040fe0f
5  clang-format     0x000000000040e2f4
6  clang-format     0x0000000000413a14
7  clang-format     0x0000000000414c76
8  clang-format     0x000000000045791a
9  clang-format     0x00000000004591f0
10 clang-format     0x000000000040bdba
11 clang-format     0x000000000040a1a8
12 libc.so.6        0x00007f7140a88830 __libc_start_main + 240
13 clang-format     0x000000000040a2b9
Segmentation fault

$ clang-format --version
clang-format version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)

**Comment by mkurdej** [comment]
FYI, since version 9, clang-format detects the BOM and errors out:
```
"d:\Program Files\LLVM-9.0.0\bin\clang-format.exe" -style=file test.cpp
error: encoding with unsupported byte order mark "UTF-16 (LE)" detected in file 'test.cpp'.
```

Until version 8, it crashed.

**Comment by owenca** [comment]
Fixed in 4ba5269f33ff.
